File: backend/src/ws_connection.py
from dispatcher.network_connection import NetworkConnection
from dispatcher.protocol.task import TaskOptions
from dispatcher.task import Task
import websocket
import json

class WSConnection(NetworkConnection):
    def __init__(self, ws: websocket.WebSocket):
        super().__init__()
        self.ws = ws

    def restoreConnection(self, ws: websocket.WebSocket):
        self.ws = ws
        self.onConnectionRestored()

    def sendTask(self, task: Task):
        clientTask = {
          'options': task.getTaskOptions(),
          'taskId': task.id,
        }
        self.ws.send(json.dumps(clientTask))
        return None 

    def abortTask(self, task: Task):
        self.ws.send(json.dumps({
          'type': 'abort',
          'task_id': task.id,
        }))
        return None 

class ClientTask:
    def __init__(self, options: TaskOptions, taskId: str):
        self.options = options
        self.taskId = taskId

File: backend/src/ws-connection.ts
import { NetworkConnection } from 'dispatcher/network-connection';
import { TaskOptions } from 'dispatcher/protocol/task';
import { Task } from 'dispatcher/task';
import { WebSocket } from 'ws';

export class WSConenction extends NetworkConnection {
  constructor(private ws: WebSocket) {
    super();
  }

  restoreConnection(ws: WebSocket) {
    this.ws = ws;
    this.onConnectionRestored();
  }

  sendTask(task: Task): Promise<void> {
    const taskOptions = task.getTaskOptions();

    const clientTask: ClientTask = {
      options: taskOptions?.standardPipeline,
      comfyOptions: taskOptions?.comfyPipeline,
      taskId: task.id,
    }

    this.ws.send(JSON.stringify(clientTask));
    return Promise.resolve();
  }

  abortTask(task: Task): Promise<void> {
    this.ws.send(JSON.stringify({
      type: 'abort',
      task_id: task.id,
    }));
    return Promise.resolve();
  }
}

export type ClientTask = {
  options?: TaskOptions['standardPipeline'],
  comfyOptions?: TaskOptions['comfyPipeline'],
  taskId: string,
};